terraform/files/bin/fixup_k8sregistry.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+# fixup_k8s_version.sh
+# Patch $2 (cluster-template.yaml) with fixed up k8s imageRepo if needed
+# This is to find old kube-proxy images on k8s.gcr.io and new ones on registry.k8s.io
+# See https://github.com/kubernetes-sigs/cluster-api/blob/main/internal/util/kubeadm/kubeadm.go#L40
+# (c) Kurt Garloff, Roman Hros, 02/2023
+# SPDX-License-Identifier: Apache-2.0
+
+if test -z "$2"; then echo "ERROR: Need clusterctl.yaml cluster-template args" 1>&2; exit 1; fi
+K8SVER=$(grep '^KUBERNETES_VERSION:' "$1" | sed 's/^KUBERNETES_VERSION: v\([0-9.]*\)/\1/')
+K8SMINOR=${K8SVER#*.}
+K8SVER=${K8SVER%%.*}${K8SMINOR%%.*}$(printf %02i 0${K8SMINOR:3:99})
```

Comment:
I found that `$(printf %02i 0${K8SMINOR:3:99})` tries to convert number to octal.
When I plugged in v1.24.8 I got `./fixup_k8sregistry.sh: line 12: printf: 08: invalid octal number` or for v1.24.9 I got `printf: 09: invalid octal number`.
For e.g. v1.24.10 I have **12408** in *$K8SVER* so it doesn't work

Reply:
Ack, this was broken, as printf interpreted numbers with leading zeros as octal numbers.
Thanks for finding this one. Fixed by [0272e8a](https://github.com/SovereignCloudStack/k8s-cluster-api-provider/pull/324/commits/0272e8a8a524606186d2b06fc9293bbe3a37299b).

Reply:
With e4f984bb60d6f99b313c428c9a77b84cf498a6eb actually.